source/MaterialXGenShader/HwLightHandler.cpp

```diff
@@ -1,25 +1,20 @@
 #include <MaterialXGenShader/HwLightHandler.h>
 #include <MaterialXGenShader/HwShaderGenerator.h>
 
+#include <iterator>
+
 namespace MaterialX
 {
 
-unsigned int HwLightHandler::getLightType(NodePtr node)
+int HwLightHandler::getLightType(NodePtr node) const
 {
-    size_t hash = std::hash<std::string>{}(node->getCategory());
-
-    // If we're on a 64-bit platform, convert to 32-bits
-    bool on64BitPlatform = sizeof(hash) == 8;
-    if (on64BitPlatform)
-    {
-        // Convert hash to 32-bits
-        return static_cast<unsigned int>(hash & 0xFFFFFFFF) ^
-               static_cast<unsigned int>((static_cast<unsigned long long>(hash) >> 32) & 0xFFFFFFFF);
-    }
-    else
+    int typeId = -1;
```

Comment:
Remove hash. Just get the position of the category string in a list of categories.
This does not have to be used as any id can be used when adding and those same ides can be used to retrieve info. This just provides convenience + also ids which are invalid for OpenGL (unless we have more than 2^32  light categoreis)